packages/runtime/src/widgets/DataGrid/DataGrid.tsx

```diff
@@ -198,13 +198,26 @@ export const DataGrid: React.FC<GridProps> = (props) => {
     return [];
   }, [values?.DataColumns]);
 
+  // handle onrow click
+  const handleOnRowClick = (
+    event: React.MouseEvent<unknown>,
+    record: unknown,
+    recordIndex?: number,
+  ): void => {
+    const targetElem = event.target as HTMLElement;
+    if (targetElem.tagName !== "TD") {
+      return;
+    }
+    onTapActionCallback(record, recordIndex);
+  };
+
   return (
     <div ref={containerRef}>
       <Table
         dataSource={namedData}
         key={resolvedWidgetId}
         onRow={(record, recordIndex) => {
-          return { onClick: () => onTapActionCallback(record, recordIndex) };
+          return { onClick: (e) => handleOnRowClick(e, record, recordIndex) };
```

Comment:
@sagardspeed2 I think you've misunderstood the issue. The issue is the event propagation from the Collapsible container. We need to wrap the Collapsible widget with a component and an event handler that stops event propagation.

Reply:
@evshi I did not understand the task in wrong way. I tried your suggested way in which we can wrap the collapsible inside an element and using that element, we need to stop the event propagation. But that thing is not working. When we add event stop propagation inside the wrapper, it also block the click event on the collapsible widget. so on the click on collapsible, it does not trigger the collapsible callback.

Reply:
@sagardspeed2 please double check what you're describing. It doesn't make sense that a container element would interfere with the event handling of its child element. I pushed a commit that resolves the issue with the approach I'm describing. Let me know how it differs from what you were seeing.